src/uct/ib/base/ib_device.c

```diff
@@ -416,16 +416,24 @@ ucs_status_t uct_ib_device_query(uct_ib_device_t *dev,
     return UCS_OK;
 }
 
+struct uct_ib_sigaction {
+    void *uapi_sa_handler;
+    unsigned long uapi_sa_flags;
+    void *uapi_sa_restorer;
+    unsigned long uapi_sa_mask;
+};
+
 static int uct_ib_device_cleanup_proc(void* arg)
 {
     uct_ib_device_nb_close_ctx *ctx = arg;
     static const char *process_name = "ucx_cleanup";
-    void *dfl                       = SIG_DFL;
+    struct uct_ib_sigaction dfl     = { SIG_DFL };
     char dummy;
     int i;
 
     for (i = 1; i <= SIGUSR2; i++) {
-        ucs_syscall_raw4(SYS_rt_sigaction, i, (long)&dfl, 0, sizeof(dfl));
+        ucs_syscall_raw4(SYS_rt_sigaction, i, (long)&dfl, 0,
+                         sizeof(dfl.uapi_sa_mask));
```

Comment:
weird

Reply:
https://elixir.bootlin.com/linux/v4.4/source/kernel/signal.c#L3307

Reply:
ic, it should be the size of sigset_t and not size of whole struct